python/lsst/pipe/analysis/utils.py

```diff
@@ -41,11 +42,53 @@
            "writeParquet", "getRepoInfo", "findCcdKey", "getCcdNameRefList", "getDataExistsRefList",
            "orthogonalRegression", "distanceSquaredToPoly", "p1CoeffsFromP2x0y0", "p2p1CoeffsFromLinearFit",
            "lineFromP2Coeffs", "linesFromP2P1Coeffs", "makeEqnStr", "catColors", "setAliasMaps",
-           "addPreComputedColumns", "addMetricMeasurement", "updateVerifyJob", "computeMeanOfFrac"]
+           "addPreComputedColumns", "addMetricMeasurement", "updateVerifyJob", "computeMeanOfFrac",
+           "persistPlots"]
 
 
 NANOJANSKYS_PER_AB_FLUX = (0*units.ABmag).to_value(units.nJy)
 
+
+def persistPlots(plotList, dataId, butler):
+    """Persist plots and parse stats yielded by the supplied generator
+
+    Parameters
+    ----------
+    plotList : `list`
+        List of generators that will yield plots
+    dataId : `dict`
+        The dataId that will be used in persisting plots
+    butler : `lsst.daf.persistence.Butler`
+        The butler that might be used in persisting the plots
+    """
+    allStats = {}
+    allStatsHigh = {}
+    n = 0
+    for plots in plotList:
+        # In this conext next is called to "prime" the generator
```

Comment:
conext -> context

So what bad occurrence is this nexting-through-a-yielded-None avoiding?

Reply:
I'm not sure I understand the question. an empty yield at the beginning of plot functions was used as a way to pause any further execution of the plotting function until a later time. This let the function context be stored in a list with all the passed in parameters et. cetera as a generator that will later be used to produce plots. Calling next causes the function to begin executing again. This could have been wrapped up into the loop as well like:
```
for plot in plots:
    if plot is None:
        continue
    ....
```
However this would hide any other place where None was expectantly being yielded. By calling next at the start it ensures that only the first yield point will not be considered. This is basically a baby step in the way that python asyncio works, but limited in scope to just want is needed to defer plotting here.